internal/controller/storageclaim_controller.go

```diff
@@ -139,7 +158,7 @@ func (r *StorageClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&csiopv1a1.ClientProfile{}, builder.WithPredicates(generationChangePredicate)).
 		Watches(
 			&extv1.CustomResourceDefinition{},
-			&handler.EnqueueRequestForObject{},
+			enqueueStorageClaimRequest,
```

Comment:
this'll not work due to usage of `c.GetName()` in predicate. Since we are sure there'll only be two storageclaims, just list and enqueue them.